kratos/containers/container_expression/specialized_container_expression.h

```diff
@@ -168,6 +168,16 @@ class SpecializedContainerExpression : public ContainerExpression<TContainerType
     ///@name Operators
     ///@{
 
+    SpecializedContainerExpression Slice(
+        const IndexType Offset,
+        const IndexType Stride) const;
+
+    SpecializedContainerExpression Reshape(const std::vector<IndexType>& rShape) const;
+
+    SpecializedContainerExpression Comb(const BaseType& rOther) const;
+
+    SpecializedContainerExpression Comb(const std::vector<typename BaseType::Pointer>& rListOfOthers) const;
```

Comment:
I'd prefer an implementation with iterators. The constructors with `std::vector` could then just call the ones with the iterators.